tedana/selection/component_selector.py

```diff
@@ -0,0 +1,560 @@
+        is set in the json file, that will take precedence. As a style rule, a
+        parameter that is the same regardless of the inputted data should be
+        defined in the decision tree json file. A parameter that is dataset specific
+        should be passed through the initialization function. Dataset specific
+        parameters that may need to be passed during initialization include:
+
+        n_echos : :obj:`int`
+            Number of echos in multi-echo fMRI data.
+            Required for kundu and minimal trees
+        n_vols: :obj:`int`
+            Number of volumes (time points) in the fMRI data
+            Required for kundu tree
+
+        An example initialization with these options would look like
+        `selector = ComponentSelector(tree, comptable, n_echos=n_echos, n_vols=n_vols)`
+        """
+
+        self.tree_name = tree
+
+        self.__dict__.update(cross_component_metrics)
+        self.cross_component_metrics = cross_component_metrics
+
+        # Construct an un-executed selector
+        self.component_table = component_table.copy()
+
+        # To run a decision tree, each component needs to have an initial classification
+        # If the classification column doesn't exist, create it and label all components
+        # as unclassified
+        if "classification" not in self.component_table:
+            self.component_table["classification"] = "unclassified"
+
+        self.tree = load_config(self.tree_name)
+        tree_config = self.tree
+
+        LGR.info("Performing component selection with " + tree_config["tree_id"])
+        LGR.info(tree_config.get("info", ""))
+        RepLGR.info(tree_config.get("report", ""))
+        RefLGR.info(tree_config.get("refs", ""))
+
+        self.tree["nodes"] = tree_config["nodes"]
+        self.necessary_metrics = set(tree_config["necessary_metrics"])
+        self.intermediate_classifications = tree_config["intermediate_classifications"]
+        self.classification_tags = set(tree_config["classification_tags"])
+        if "used_metrics" not in self.tree.keys():
+            self.tree["used_metrics"] = set()
+        else:
+            self.tree["used_metrics"] = set(self.tree["used_metrics"])
+
+        if status_table is None:
+            self.component_status_table = self.component_table[
+                ["Component", "classification"]
+            ].copy()
+            self.component_status_table = self.component_status_table.rename(
+                columns={"classification": "initialized classification"}
+            )
+            self.start_idx = 0
+        else:
+            # Since a status table exists, we need to skip nodes up to the
+            # point where the last tree finished
+            self.start_idx = len(tree_config["nodes"])
+            LGR.info(f"Start is {self.start_idx}")
+            self.component_status_table = status_table
+
+    def select(self):
```

Comment:
Thinking this over. Assuming we want to keep the option open to load a tree and then create a component table from the `required_metrics` in the tree then we wouldn't want those parameters in `init`. We might not want them in `select` because `select` can be called on a tree with existing information and an existing component_table (i.e. by `tedana_reclassify`). To account for multiple use cases, it might make sense to have 3 steps & 4 functions: 
`init`
`load_component_table` OR `calc_component_table`
`selection`
I think that would be a better change to make when we map out those two intermediate functions & how they interact with workflows. Until then, I think it would make sense to better separate `component_table`, `cross_component_metrics`, and `status_table` within `init` OR `select` so that a future change would be easier.

Reply:
Dev call consensus was to table this until after the PR is merged. (If I have the time I might clean things up so it will be easier in the future). Let's keep this thread open until I either make the partial change we we open an issue so that this doesn't slip.